crates/rpc-types-eth/src/filter.rs

```diff
@@ -14,47 +15,50 @@ use itertools::{
     Itertools,
 };
 
-/// Helper type to represent a bloom filter used for matching logs.
-#[derive(Debug, Default)]
-pub struct BloomFilter(Vec<Bloom>);
+/// FilterSet is a set of values that will be used to filter logs.
+#[derive(Clone, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+pub struct FilterSet<T: Eq + Hash> {
+    set: HashSet<T>,
 
-impl From<Vec<Bloom>> for BloomFilter {
-    fn from(src: Vec<Bloom>) -> Self {
-        Self(src)
-    }
+    #[cfg(feature = "std")]
+    #[cfg_attr(feature = "serde", serde(skip, default))]
+    bloom: std::sync::OnceLock<Bloom>,
```

Comment:
I think we can use once_cell here and get rid of the std features like this once_cell should be already part of the dep graph anyway

Reply:
`once_cell`'s no_std support is a little wonky. do you use it with race or critical section?